projects/igniteui-angular/src/lib/grids/grid/grid.master-detail.spec.ts

```diff
@@ -43,125 +49,136 @@ describe('IgxGrid Master Detail #grid', () => {
-            expect(firstRowIcon.innerText).toEqual(EXPANDED_ICON_NAME);
-            expect(firstRowDetail.querySelector('.addressArea').innerText).toEqual('Obere Str. 57');
-        }));
+            expect(firstRowIconName).toEqual(EXPANDED_ICON_NAME);
+            expect(getDetailAddressText(firstRowDetail)).toEqual('Obere Str. 57');
+        });
 
-        it('Should render a detail view with dynamic elements and they should be clickable/focusable.', (async() => {
-            await GridFunctions.toggleMasterRowByClick(fix, grid.rowList.first, DEBOUNCETIME);
+        it('Should render a detail view with dynamic elements and they should be clickable/focusable.', async() => {
+            GridFunctions.toggleMasterRow(fix, grid.rowList.first);
+            await wait();
+            fix.detectChanges();
 
-            const firstRowDetail = GridFunctions.getMasterRowDetail(grid.rowList.first);
-            const checkboxElem = firstRowDetail.querySelector('igx-checkbox');
-            const checkboxPos = checkboxElem.getBoundingClientRect();
-            const inputElem = firstRowDetail.querySelector('.igxInputGroup');
-            const inputElemPos = inputElem.getBoundingClientRect();
+            const firstDetail = GridFunctions.getMasterRowDetailDebug(fix, grid.rowList.first);
+            const checkboxElem = firstDetail.query(By.directive(IgxCheckboxComponent));
+            const checkboxPos = checkboxElem.nativeElement.getBoundingClientRect();
+            const inputElem = firstDetail.query(By.directive(IgxInputGroupComponent));
+            const inputElemPos = inputElem.nativeElement.getBoundingClientRect();
 
             const tracedCheckbox: any =
                 document.elementFromPoint(checkboxPos.left + checkboxPos.height / 2, checkboxPos.top + checkboxPos.height / 2);
             const tracedInput: any =
                 document.elementFromPoint(inputElemPos.left + inputElemPos.height / 2, inputElemPos.top + inputElemPos.height / 2);
 
-            checkboxElem.querySelector('.igx-checkbox__input').click();
+            checkboxElem.componentInstance.nativeCheckbox.nativeElement.click();
             fix.detectChanges();
 
-            expect(checkboxElem.contains(tracedCheckbox)).toBeTruthy();
-            expect(checkboxElem.firstElementChild.getAttribute('aria-checked')).toEqual('true');
+            expect(checkboxElem.nativeElement.contains(tracedCheckbox)).toBeTruthy();
+            expect(checkboxElem.componentInstance.checked).toBeTruthy();
 
             UIInteractions.clickElement(inputElem);
             fix.detectChanges();
 
-            expect(inputElem.contains(tracedInput)).toBeTruthy();
+            expect(inputElem.nativeElement.contains(tracedInput)).toBeTruthy();
             expect(document.activeElement).toEqual(tracedInput);
-        }));
+        });
 
         it(`Should persist state of rendered templates, such as expansion state of expansion panel,
```

Comment:
acync for this test can be removed

Reply:
Async is required when using navigateTo since it scrolls the grid.